File: BisectionMethod.cpp
#include <iostream>
#include <cmath>

double bisection(double a, double b, const double ACCURACY);
double f(double x);

int main() {
    std::cout << bisection(1, 10, 0.00001) << std::endl;
}

double bisection(double a, double b, const double ACCURACY) {
    double z = (a + b) / 2;

    if (fabs(f(z)) < ACCURACY)
        return z;
    while (fabs(f(z)) > ACCURACY) {
        if (f(a) * f(z) < 0)
            b = z;
        else
            a = z;

        z = (a + b) / 2;

        if (fabs(a - b) < ACCURACY)
            return z;
    }
    return z;
}

double f(double x) {
    return x * x * x + 3 * x - 5;
}

File: BisectionMethod.py
def f(x):
    return x * x * x + 3 * x - 5


def Bisection(a, b, ACCURACY):
    z = (a + b) / 2
    if abs(f(z)) < ACCURACY:
        return z
    while abs(f(z)) > ACCURACY:
        if f(a) * f(z) < 0:
            b = z
        else:
            a = z
        z = (a + b) / 2
        if abs(a - b) < ACCURACY:
            return z
    return z


print(Bisection(0, 10, 0.00001))